python/tests/server/test_worker.py

```diff
@@ -670,7 +720,19 @@ def await_predict(self, state: PredictState):
     )
     def cancel(self, state: PredictState):
         self.worker.cancel(tag=state.tag)
-        assert self.child.cancel_sent
+
+        if state.canceled:
+            # if this has previously been canceled, we expect no Cancel event
+            # sent to the child
+            assert not self.child_events.poll(timeout=0.1)
```

Comment:
I think re https://github.com/replicate/cog/pull/2057#discussion_r1846392516 I'm actually more in favour of doing less asserting here rather than more. The code here is mainly to ensure we stay synced up with the child connection, not to make actual assertions - that should happen on the outside of the black box, as it were.